wayland-display-core/src/wayland/protocols/wl_drm.rs

```diff
@@ -156,29 +156,30 @@ where
                     return;
                 }
 
-                let mut dma = Dmabuf::builder((width, height), format, DmabufFlags::empty());
-                dma.add_plane(name, 0, offset0 as u32, stride0 as u32, Modifier::Invalid);
+                let mut dma = Dmabuf::builder((width, height), format, Modifier::Invalid, DmabufFlags::empty());
+                dma.add_plane(name, 0, offset0 as u32, stride0 as u32);
                 match dma.build() {
                     Some(dmabuf) => {
-                        match state.dmabuf_imported(&data.dmabuf_global, dmabuf.clone()) {
-                            Ok(_) => {
-                                // import was successful
-                                data_init.init(id, dmabuf);
-                            }
-
-                            Err(ImportError::InvalidFormat) => {
-                                drm.post_error(
-                                    wl_drm::Error::InvalidFormat,
-                                    "format and plane combination are not valid",
-                                );
-                            }
-
-                            Err(ImportError::Failed) => {
-                                // Buffer import failed. The protocol documentation heavily implies killing the
-                                // client is the right thing to do here.
-                                drm.post_error(wl_drm::Error::InvalidName, "buffer import failed");
-                            }
-                        }
+                        todo!("How can I make a ImportNotifier? Constructor is private..");
```

Comment:
I've changed  `DmabufHandler::dmabuf_imported` so that it takes an `ImportNotifier` in input instead of returning the result like it used to; it seems that I can't find a way to create it though; the constructor is private, what am I missing?

Reply:
Yeah, sadly this is the case, so we can't piggy back on the existing `DmabufHandler` implementation any more.

So we either need our own trait and implement that as well (e.g. see how cosmic-comp does this now: https://github.com/pop-os/cosmic-comp/blob/master/src/wayland/protocols/drm.rs#L47) or just drop the `D` and the split between `protocols` and `handler` and just hard-code everything for our state struct, so that we can just call `state.renderer.import_dmabuf` directly here.